### PR TITLE
ACH Integration

### PR DESCRIPTION
**Pull request by paul-gene**
This PR adds ACH (Automated Clearing House) Direct Debit as a payment option on the checkout.

Currently uses a basic network check for verification.

### CODE REVIEW

File: Model/Ach/Ui/ConfigProvider.php [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+    /**
+     * @return mixed
+     */
+    public function getStoreName()
+    {
+        $storeName = $this->scopeConfig->getValue(
+            self::CONFIG_STORE_NAME,
+            ScopeInterface::SCOPE_STORE
+        );
+        return $storeName ?: 'this store';
```
Comment (rich1990) [inline Model/Ach/Ui/ConfigProvider.php]: this store?

File: Model/Ach/Ui/ConfigProvider.php [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+            }
+
+            $this->clientToken = $this->adapter->generate($params);
+        }
+
+        return $this->clientToken;
+    }
+
+    /**
+     * @return mixed
```
Comment (rich1990) [inline Model/Ach/Ui/ConfigProvider.php]: isnt returning a string?

File: Model/Ach/Ui/ConfigProvider.php (hunk 1 above)
Comment (paul-gene) [reply Model/Ach/Ui/ConfigProvider.php]: So if no store name is set, I need some form of basic placeholder name to go into the mandate text string. Wasn't sure what else to put :D

Comment (rich1990) [reply Model/Ach/Ui/ConfigProvider.php]: you could use this:
```
return $this->_scopeConfig->getValue(
        'general/store_information/name',
        \Magento\Store\Model\ScopeInterface::SCOPE_STORE
    );
```
will return the store name

Comment (paul-gene) [reply Model/Ach/Ui/ConfigProvider.php]: That's what I'm doing, but sometimes it's empty :)

Comment (rich1990) [reply Model/Ach/Ui/ConfigProvider.php]: mh, let me think if there is another way

Comment (rich1990) [reply Model/Ach/Ui/ConfigProvider.php]: alternatively you could take the email sender name (but that could be empty too) or even better the domain name as last resort
i am just thinking to make it more dynamic as possible

Comment (paul-gene) [reply Model/Ach/Ui/ConfigProvider.php]: Ooh, the domain could be a shout! That is practically a certain that is has a value, correct?

Comment (rich1990) [reply Model/Ach/Ui/ConfigProvider.php]: yes correct! that's need to be set ;)

File: view/frontend/web/js/view/payment/method-renderer/ach.js [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+                    return;
+                }
+
+                fullScreenLoader.startLoader();
+
+                var self = this;
+
+                var billingAddress = quote.billingAddress();
+
+                let regionCode;
```
Comment (brightonmike) [inline view/frontend/web/js/view/payment/method-renderer/ach.js]: Are we okay with the lack of opera mini support for let? Probably, but just wanted to check

Comment (paul-gene) [reply view/frontend/web/js/view/payment/method-renderer/ach.js]: I think we can handle any backlash from Opera Mini fanatics ;)

**Pull request merged**